src/game_battlealgorithm.cpp

```diff
@@ -1134,6 +1168,19 @@ bool Game_BattleAlgorithm::Skill::Execute() {
 				conditions.push_back(Data::states[i]);
 			}
 		}
+
+		// Attribute resistance / weakness + an attribute selected + can be modified
+		if (!success && skill.affect_attr_defence) {
+			for (int i = 0; i < skill.attribute_effects.size(); i++) {
+				if (skill.attribute_effects[i] && GetTarget()->GetShiftAttributeRate(i + 1) * (IsPositive() ? 1 : -1) != 1) {
```

Comment:
> shift attrib

I find this ugly to read. Maybe there should be instead of GetShiftAttributeRate just a CanShiftAttributeRate which probes if a shift would be possible and returns a bool...